src/firefly/js/ui/tap/TableSelectViewPanel.jsx

```diff
@@ -271,202 +271,205 @@ function AdqlUI({serviceUrl}) {
-                                    this.loadColumns(serviceUrl, schemaName, selectedTapTable);
-                                }}
-                            />
-                        </div>
-                    </div>
-                </div>
-
-                <div className='TapSearch__section' style={{flexDirection: 'column', flexGrow: 1}}>
-                    <div style={{ display: 'inline-flex', width: 'calc(100% - 3px)', justifyContent: 'space-between'}}>
-                        <div className='TapSearch__section--title'>4. Select Constraints <HelpIcon helpId={tapHelpId('constraints')}/> </div>
-                        <TableColumnsConstraintsToolbar key={tableName}
-                                                        tableName={tableName}
-                                                        columnsModel={columnsModel}
-                        />
-                    </div>
-                    <div className='expandable'>
-                        <SplitPane split='vertical' maxSize={splitMax} mixSize={20} defaultSize={splitDef}>
-                            <SplitContent>
-                                {columnsModel ?  <TableSearchMethods columnsModel={columnsModel} initArgs={initArgs}/>
-                                    : <div className='loading-mask'/>
-                                }
-                            </SplitContent>
-                            <SplitContent>
-                                { columnsModel ?
-                                    <TableColumnsConstraints
-                                        key={tableName}
-                                        fieldKey={'tableconstraints'}
-                                        columnsModel={columnsModel}
-                                    />
-                                    : <div className='loading-mask'/>
-
-                                }
-                            </SplitContent>
-                        </SplitPane>
-                    </div>
-                </div>
-
-            </Fragment>
-        );
-    }
+    useEffect(() => {
+        stateRef.current = state;
+    }, [state]);
+    return [state, stateRef, setState];
+}
 
-    loadSchemas(serviceUrl, schemaName=undefined, tableName=undefined) {
-        this.setState({error: undefined, schemaOptions: undefined, schemaName: undefined,
-            tableOptions: undefined, tableName: undefined, columnsModel: undefined});
+function BasicUI(props) {
+    const tapFluxState = getTapBrowserState();
+    // const isObsTap = props.obsTap !== undefined ? props.obsTap : false;
+    const [error, setError] = useState(undefined);
+    const [serviceUrl, serviceUrlRef, setServiceUrl] = useStateRef(tapFluxState.serviceUrl || props.serviceUrl);
+    const [schemaName, schemaRef, setSchemaName] = useStateRef(tapFluxState.schemaName || props.initArgs.schema);
+    const [tableName, tableRef, setTableName] = useStateRef(tapFluxState.tableName || props.initArgs.table);
```

Comment:
Existing logic relies on local props/states as well as external states from flux/store made this a problem more complicated than it needs to be.  As it is now, the use of useStateRef to solve this problem is a good one.